packages/analytics-js-integrations/src/integrations/MoEngage/browser.js

```diff
@@ -80,45 +107,103 @@ class MoEngage {
-      this.moeClient.track_event(event);
+      window.Moengage.track_event(event);
     }
   }
 
   reset() {
-    // reset the user id
     this.initialUserId = this.analytics.getUserId();
-    this.moeClient.destroy_session();
+    window.Moengage.destroy_session();
   }
 
-  identify(rudderElement) {
+  identifyOld(rudderElement) {
     const { userId, context } = rudderElement.message;
     let traits = null;
     if (context) {
       traits = context.traits;
     }
+
     // check if user id is same or not
     if (this.initialUserId !== userId) {
       this.reset();
     }
     // if user is present map
     if (userId) {
-      this.moeClient.add_unique_user_id(userId);
+      window.Moengage.add_unique_user_id(userId);
     }
 
     // track user attributes : https://docs.moengage.com/docs/tracking-web-user-attributes
     if (traits) {
       each((value, key) => {
         // check if name is present
         if (key === 'name') {
-          this.moeClient.add_user_name(value);
+          window.Moengage.add_user_name(value);
+        }
+        if (Object.hasOwn(traitsMap, key)) {
+          const method = `add_${traitsMap[key]}`;
+          window.Moengage[method](value);
+        } else {
+          window.Moengage.add_user_attribute(key, value);
+        }
+      }, traits);
+    }
+  }
+
+  identify(rudderElement) {
+    if (!this.identityResolution) {
+      this.identifyOld(rudderElement);
+      return;
+    }
+
+    const { userId, context } = rudderElement.message;
+    let traits = null;
+    if (context) {
+      traits = context.traits;
+    }
+
+    // We are destroying the session if userId is changed or initialUserId is not empty and userId is empty
+    // This is a log out scenario
+    if (
+      (this.initialUserId !== '' && this.initialUserId !== userId) ||
+      (this.initialUserId !== '' && userId === '')
+    ) {
+      this.reset();
+    }
+
+    // If initialUserId is empty, set it to the current userId this happens when an anonymous user loggedIn for the first time
+    if (this.initialUserId === '' && userId) {
+      this.initialUserId = userId;
+    }
+
+    const userAttributes = {};
+    each((value, key) => {
+      if (Object.hasOwn(identifyUserPropertiesMap, key)) {
+        const method = identifyUserPropertiesMap[key];
+        userAttributes[method] = value;
+      }
+    }, traits);
+
+    const payload = {
+      anonymousId: this.analytics.getAnonymousId(),
+      ...(userId && { uid: userId }),
+      ...userAttributes,
+    };
+    window.Moengage.identifyUser(payload);
+
+    // track user attributes : https://developers.moengage.com/hc/en-us/articles/360061114832-Web-SDK-User-Attributes-Tracking
+    if (traits) {
+      each((value, key) => {
+        // check if name is present
+        if (key === 'name') {
+          window.Moengage.add_user_name(value);
         }
-        if (Object.prototype.hasOwnProperty.call(traitsMap, key)) {
+        if (Object.hasOwn(traitsMap, key)) {
```

Comment:
identifyUserPropertiesMap already covers most of the properties from traitsMap under userAttributes 
Is there a need to do this again or just focus on custom attributes ?